### ISSUE TITLE
【Need Help】Screen orientation Upside Down

### ISSUE THREAD

**Issue by CoWillChen** [jiulongw/swift-unity#93]
I'found a strange problem that the screen orientation will be upside down sometimes after integrated into the swift project.
Key Code: 
` GUI.Label(new Rect(50, 300, 600, 300),"d:"+ Input.deviceOrientation.ToString() + " s:" + Screen.orientation.ToString());`

As the attached video shows, if I rotate the device by anti-clockwise (Top Left, Home Right)，Input.deviceOrientation tells me the orientation is LandscapeLeft, but Screen.orientation tells LandscapeRight. Then I rotate it back to portrait, Screen.orientation tells Portrait UpsideDown.
More Details:
1. If I just export the iOS project(without XcodePostBuild.cs) from Unity, and build it , there's no issue.
2. With XcodePostBuild.cs, if I build the swift project on iOS10 or lower, no issue, but on iOS 11 or higher, the issue happen.
3. The issue will happen when I rotate the device by anticlockwise, however, clockwise is fine.
4. Tested on Unity 2017.3.1 , 2018.1.0 , 2018.2.0

 Any Help? Thanks.

---Chinese Version --- 
英文比较烂，加个中文注释吧。。。
一旦将Unity的工程集成到swift工程，会出现Screen.orientation翻转的情况，逆时针转动手机，就会出现问题，例如附件中视频中显示的，我把手机转向左侧，Input.deviceOrientation 显示landscapeLeft, 但是Screen.orientation显示是landscapeRight，我再转回来，一个是portrait，一个已经是portraitUpsideDown了。但是向右转，就不会出现问题。
这个问题仅出现在集成到本地工程中，单纯从unity导出（不加XcodePostBuild.cs文件）就没问题。
如果加了集成到本地swift工程中，build的目标是iOS11以下的平台也没问题，一旦iOS11或以上，就会出现上述情况。 大神能找到什么头绪吗？谢谢
[IMG_3125.MOV.zip](https://github.com/jiulongw/swift-unity/files/2230333/IMG_3125.MOV.zip)

**Comment by CoWillChen** [comment]
Simply make the ViewController to  be a subclass of UnityViewControllerBase.

**Comment by jiulongw** [comment]
我很久没有做相关的项目了，但这个问题听上去很有意思。你能打包一个最小的可以重现的工程给我看看吗？

感觉应该是xcode里关于orientation的设置的问题...

**Comment by CoWillChen** [comment]
非常感谢大神热心帮助，我用你的demo做了些修改，重现了这个情况，因为包比较大，我直接传百度云了：
链接:https://pan.baidu.com/s/1gqtzavYkwS3hbK8lnMJC3g  密码:76tg

PS: 只有build到iOS11及以上的iPhone上才会出现，逆时针一转就有问题。

**Comment by CoWillChen** [comment]
要是工程重现上有什么问题，您可以直接发我邮件(cowill@foxmail.com) or QQ(373976541)，方便快速交流。近期上线一个活动要用到orientation的转向，感谢大神帮助！

**Comment by CoWillChen** [comment]
我发现是Classes -- UI -- UnityViewControllerBase+iOS.mm里的 - static void ViewWillTransitionToSize_DefaultImpl(id self_, SEL _cmd, CGSize size, id<UIViewControllerTransitionCoordinator> coordinator) 这个函数的问题。
它在取当前的转向curOrient时取到的其实就是转好后的目标orientation，结果之后又计算了一步newOrient= OrientationAfterTransform(curOrient, [coordinator targetTransform]); 再去做一次转向 就出错了，我暂时把这个函数直接return掉，由swift工程通知UnityAppController转向，问题就修复了。
但想不明白为何 ViewWillTransitionToSize_DefaultImpl 明明是在Will阶段触发的，取到的curOrient却是已经转好的，导致了错误。

**Comment by aiursrage2k** [comment]
Hello can you please post the code how you solved this thanks.

**Comment by Kadrius** [comment]
I managed to make this working with a little help of google translator :D.
For any english speaker out there, the way I fixed this is by changing the function **_viewWillTransitionToSize_** in the file **_UnityViewControllerBase+iOS_** located un Classes/UI. I had to change this line:

`ScreenOrientation newOrient = OrientationAfterTransform(curOrient, [coordinator targetTransform]);`

with this one:

`ScreenOrientation newOrient = curOrient;`

For some reason the curOrient is already the correct orientation and if we do the transformation Vuforia will rotate 180 degrees the camera rendering.
